opsdroid/connector/rocketchat/__init__.py

```diff
@@ -0,0 +1,190 @@
+
+        Args:
+            opsdroid (OpsDroid): An instance of opsdroid core.
+            response (dict): Response returned by aiohttp.Client.
+
+        """
+        if response['messages']:
+            message = Message(
+                response['messages'][0]['msg'],
+                response['messages'][0]['u']['username'],
+                self.default_room,
+                self)
+            _LOGGER.debug("Received message from Rocket.Chat %s",
+                          response['messages'][0]['msg'])
+
+            await opsdroid.parse(message)
+            self.latest_update = response['messages'][0]['ts']
+
+    async def _get_message(self, opsdroid):
+        """Connect to the API and get messages.
+
+        This method will only listen to either a channel or a
+        private room called groups by Rocket.Chat. If a group
+        is specified in the config then it takes priority
+        over a channel.
+
+        Args:
+            opsdroid (OpsDroid): An instance of opsdroid.core.
+
+        """
+        if self.group:
+            url = self.build_url('groups.history?roomName={}'.format(
+                self.group))
+            self.default_room = self.group
+        else:
+            url = self.build_url('channels.history?roomName={}'.format(
+                self.default_room))
+
+        if self.latest_update:
+            url += '&oldest={}'.format(self.latest_update)
+
+        async with aiohttp.ClientSession() as session:
+            resp = await session.get(url,
+                                     headers=self.headers)
+
+            if resp.status != 200:
+                _LOGGER.error("Rocket.Chat error %s, %s",
+                              resp.status, resp.text)
+                self.listening = False
+            else:
+                json = await resp.json()
+                await self._parse_message(opsdroid, json)
+
+    async def listen(self, opsdroid):
+        """Listen for and parse new messages.
+
+        The method will sleep asynchronously at the end of
+        every loop. The time can either be specified in the
+        config.yaml with the param update-interval - this
+        defaults to 1 second.
+
+        If the channel didn't get any new messages opsdroid
+        will still call the REST API, but won't do anything.
+
+        Args:
+            opsdroid (Opsdroid): An instance of opsdroid core.
+
+        """
+        while self.listening:
+            await self._get_message(opsdroid)
+            await asyncio.sleep(self.update_interval)
+
+    async def respond(self, message, room=None):
+        """Respond with a message.
+
+        Args:
+            message (object): An instance of Message
+            room (string, optional): Name of the room to respond to.
+
+        """
+        _LOGGER.debug("Responding with: %s", message.text)
+        async with aiohttp.ClientSession() as session:
+            data = {}
+            data['channel'] = "#{}".format(self.default_room)
```

Comment:
Should this always be responding to the default room? What about the room that is passed into this method?

Reply:
The reason why I added the default is because this is changed depending on group/channel in the config. I can try to just use the room passed to this method, cant remember if I tried it

Reply:
The room that gets passed in will be the same room you set when you construct the `Message` object. So using that should be fine.